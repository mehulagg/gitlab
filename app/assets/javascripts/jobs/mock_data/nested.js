export default {
  lines: [
    {
      offset: 0,
      content: [{ text: 'Hello' }],
    },
    {
      offset: 5,
      content: [{ text: 'foo' }],
      sections: ['prepare-script'],
      section_header: true,
    },
    {
      offset: 52,
      content: [{ text: 'bar' }],
      sections: ['prepare-script', 'prepare-script-nested'],
      section_header: true,
      section_duration: "00:02",
    },
    {
      offset: 80,
      content: [{ text: 'this is a collapsible nested section' }],
      sections: ['prepare-script', 'prepare-script-nested'],
    },
    {
      offset: 106,
      content: [],
      sections: ['prepare-script'],
      section_duration: "00:03",
    },
    {
      offset: 155,
      content: [],
    },
  ],
};
