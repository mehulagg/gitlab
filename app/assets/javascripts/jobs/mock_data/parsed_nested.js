export default {
  lines: [
    {
      offset: 0,
      content: [{ text: 'Hello' }],
      lineNumber: 0,
    },
    {
      offset: 5,
      section_header: true,
      isHeader: true,
      isClosed: true,
      line: {
        content: [{ text: 'foo' }],
        sections: ['prepare-script'],
        lineNumber: 1,
      },
      section_duration: "00:03",
      lines: [
        {
          section_header: true,
          section_duration: "00:02",
          isHeader: true,
          isClosed: true,
          line: {
            offset: 52,
            content: [{ text: 'bar' }],
            sections: ['prepare-script', 'prepare-script-nested'],
            lineNumber: 2,
          },
          lines: [
            {
              offset: 80,
              content: [{ text: 'this is a collapsible nested section' }],
              sections: ['prepare-script', 'prepare-script-nested'],
              lineNumber: 3.
            }
          ]
        }
      ]
    },
  ],
};
