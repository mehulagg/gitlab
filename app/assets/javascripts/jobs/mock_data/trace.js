// Using https://gitlab.com/gitlab-org/gitlab-ce/-/jobs/278912562

export default {
  id: 278912562,
  status: 'failed',
  complete: true,
  state:
    'eyJvZmZzZXQiOjQ5NzU5NCwibl9vcGVuX3RhZ3MiOjAsImZnX2NvbG9yIjpudWxsLCJiZ19jb2xvciI6bnVsbCwic3R5bGVfbWFzayI6MCwic2VjdGlvbnMiOltdLCJsaW5lbm9faW5fc2VjdGlvbiI6MTJ9',
  append: false,
  truncated: false,
  offset: 0,
  size: 497594,
  total: 497594,

  lines: [
    // {
    //   offset: 1000,
    //   content: [{ text: 'Running with gitlab-runner 12.1.0 (de7731dd)' }],
    // },
    // {
    //   offset: 1001,
    //   content: [{ text: ' on docker-auto-scale-com 8a6210b8' }],
    // },
    // {
    //   offset: 1002,
    //   content: [
    //     {
    //       text:
    //         'Using Docker executor with image dev.gitlab.org:5005/gitlab/gitlab-build-images:ruby-2.6.3-golang-1.11-git-2.22-chrome-73.0-node-12.x-yarn-1.16-postgresql-9.6-graphicsmagick-1.3.33',
    //     },
    //   ],
    //   section: 'prepare-executor',
    //   section_header: true,
    // },
    // {
    //   offset: 1003,
    //   content: [{ text: 'Starting service postgres:9.6.14 ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1004,
    //   content: [{ text: 'Pulling docker image postgres:9.6.14 ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1005,
    //   content: [
    //     {
    //       text: `Using docker image sha256:ac400042d32f8e146477ae5b3528f1e4232fc8e83b2fd9fe0cbe08900e45bae1 for
    //   postgres:9.6.14 ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1006,
    //   content: [{ text: 'Starting service redis:alpine ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1007,
    //   content: [{ text: 'Pulling docker image redis:alpine ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1008,
    //   content: [
    //     {
    //       text: `Using docker image sha256:ed7d2ff5a6232b43bdc89a2220ed989f532c3794422aa2a86823b8bc62e71447 for
    //   redis:alpine ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1009,
    //   content: [{ text: 'Waiting for services to be up and running...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1010,
    //   content: [
    //     {
    //       text: `Pulling docker image
    //   dev.gitlab.org:5005/gitlab/gitlab-build-images:ruby-2.6.3-golang-1.11-git-2.22-chrome-73.0-node-12.x-yarn-1.16-postgresql-9.6-graphicsmagick-1.3.33
    //   ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1011,
    //   content: [
    //     {
    //       text: `Using docker image sha256:f5b71ccc7a53408ea88959433baf23f7d716bb431d0475a4f3d9ba213d7841c1 for
    //   dev.gitlab.org:5005/gitlab/gitlab-build-images:ruby-2.6.3-golang-1.11-git-2.22-chrome-73.0-node-12.x-yarn-1.16-postgresql-9.6-graphicsmagick-1.3.33
    //   ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1012,
    //   content: [],
    //   section: 'prepare-executor',
    //   section_footer: true,
    //   section_duration: '10:00'
    // },
    // {
    //   offset: 1013,
    //   content: [
    //     {
    //       text: `Running on runner-8a6210b8-project-13083-concurrent-0 via
    //   runner-8a6210b8-gsrm-1566561403-37c881a0...`,
    //     },
    //   ],
    //   section: 'prepare-script',
    //   section_header: true,
    // },

    // {
    //   offset: 1014,
    //   content: [
    //     { text: 'Fetching changes with git depth set to 20..', style: 'term-fg-l-green term-bold' },
    //   ],
    //   section: 'get-sources',
    //   section_header: true,
    // },
    // {
    //   offset: 1015,
    //   content: [
    //     {
    //       text: 'Reinitialized existing Git repository in /builds/gitlab-org/gitlab-ce/.git/ ',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1016,
    //   content: [
    //     {
    //       text: 'From https://gitlab.com/gitlab-org/gitlab-ce',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1017,
    //   content: [
    //     {
    //       text: ' * [new branch] 66454-create-feature-flag -&gt; origin/66454-create-feature-flag',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1018,
    //   content: [
    //     {
    //       text: 'Checking out 061606b4 as 66454-create-feature-flag...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1019,
    //   content: [
    //     {
    //       text: 'Removing .gitlab_shell_secret',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1020,
    //   content: [
    //     {
    //       text: 'Removing .gitlab_workhorse_secret',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1021,
    //   content: [
    //     {
    //       text: 'Removing config/database_geo.yml',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1022,
    //   content: [
    //     {
    //       text: 'Removing config/gitlab.yml',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1023,
    //   content: [
    //     {
    //       text: 'Removing config/redis.cache.yml',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1024,
    //   content: [
    //     {
    //       text: 'Skipping Git submodules setup',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1025,
    //   content: [],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1026,
    //   content: [],
    //   section: 'get-sources',
    //   section_footer: true,
    //   section_duration: '30:00'
    // },
    // {
    //   offset: 1027,
    //   content: [
    //     {
    //       text: 'Checking cache for debian-stretch-ruby-2.6.3-node-12.x-3...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['restore-cache'],
    //   section_header: true,
    // },
    // {
    //   offset: 1028,
    //   content: [],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1029,
    //   content: [
    //     {
    //       text:
    //         'Downloading cache.zip from https://storage.googleapis.com/gitlab-com-runners-cache/project/13083/debian-stretch-ruby-2.6.3-node-12.x-3',
    //     },
    //   ],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1030,
    //   content: [],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1031,
    //   content: [
    //     {
    //       text: 'Successfully extracted cache',
    //       style: 'term-fg-l-green term-bold ',
    //     },
    //   ],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1032,
    //   content: [],
    //   sections: ['restore-cache'],
    //   section_footer: true,
    //   section_duration: '00:10'
    // },
    // {
    //   offset: 1033,
    //   content: [
    //     {
    //       text: 'Downloading artifacts for compile-assets pull-cache (278912482)...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    //   section_header: true,
    // },
    // {
    //   offset: 1034,
    //   content: [],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1035,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok ',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //     {
    //       text: ' id',
    //     },
    //     {
    //       text: '=278912482 responseStatus',
    //     },
    //     {
    //       text: '=200 OK token',
    //     },
    //     {
    //       text: '=DUNW1asp',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1036,
    //   content: [
    //     {
    //       text: 'Downloading artifacts for setup-test-env (278912484)...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1037,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok  id=278912484 responseStatus=200 OK token=gdMnE5zT',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1038,
    //   content: [
    //     {
    //       text: `WARNING: tmp/tests/gitlab-shell/.gitlab_shell_secret: chmod
    //       tmp/tests/gitlab-shell/.gitlab_shell_secret: no such file or directory (suppressing
    //       repeats)`,
    //       style: 'term-fg-yellow',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1039,
    //   content: [
    //     {
    //       text: 'Downloading artifacts for retrieve-tests-metadata (278912485)...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1040,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok id=278912485 responseStatus=200 OK token=rCtxUhsy',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1041,
    //   content: [],
    //   sections: ['download-artifacts'],
    //   section_footer: true,
    //   section_duration: '01:00:58'
    // },
    // {
    //   offset: 1042,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok ',
    //     },
    //     {
    //       text: ' id',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //     {
    //       text: '=278912484 responseStatus=200 OK token=gdMnE5zT',
    //     },
    //   ],
    //   sections: 'download-artifacts',
    // },
    {
      offset: 1043,
      content: [{ text: 'Hello' }],
    },
    {
      offset: 5,
      content: [{ text: 'foo' }],
      section: 'prepare-script',
      section_header: true,
    },
    {
      offset: 52,
      content: [{ text: 'bar' }],
      section: 'prepare-script-nested',
      section_header: true,
    },
    {
      offset: 80,
      content: [{ text: 'this is a collapsible nested section' }],
      section: 'prepare-script-nested',
    },
    {
      offset: 90,
      section: 'prepare-script-nested',
      section_duration: '00:02',
    },
    {
      offset: 106,
      content: [],
      section: 'prepare-script',
      section_duration: '00:03',
    },
    {
      offset: 107,
      content: [],
      section_duration: '00:04',
    },
    {
      offset: 155,
      content: [],
    },
  ],
};
