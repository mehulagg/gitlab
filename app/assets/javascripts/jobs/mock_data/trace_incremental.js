// Using https://gitlab.com/gitlab-org/gitlab-ce/-/jobs/278912562

export default {
  id: 278912562,
  status: 'failed',
  complete: true,
  state:
    'eyJvZmZzZXQiOjQ5NzU5NCwibl9vcGVuX3RhZ3MiOjAsImZnX2NvbG9yIjpudWxsLCJiZ19jb2xvciI6bnVsbCwic3R5bGVfbWFzayI6MCwic2VjdGlvbnMiOltdLCJsaW5lbm9faW5fc2VjdGlvbiI6MTJ9',
  append: false,
  truncated: false,
  offset: 0,
  size: 497594,
  total: 497594,

  lines: [
    {
      offset: 1042,
      content: [
        {
          text:'Not downloading anything ',
        },
        {
          text: ' id',
          style: 'term-fg-l-green term-bold',
        },
        {
          text: '=278912484 responseStatus=200 OK token=gdMnE5zT',
        },
      ],
      sections: ['download-artifacts'],
    },

    {
      offset: 1043,
      content: [
        {
          text: 'A filipa Ã© fixe e o resto que se lixe ',
        },
        {
          text: ' id',
          style: 'term-fg-l-green term-bold',
        },
        {
          text: '=278912484 responseStatus=200 OK token=gdMnE5zT',
        },
      ],
    },
  ],
};
