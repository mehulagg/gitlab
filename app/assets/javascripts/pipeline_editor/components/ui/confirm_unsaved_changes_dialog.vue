<script>
export default {
  props: {
    hasUnsavedChanges: {
      type: Boolean,
      required: true,
    },
  },
  created() {
    window.addEventListener('beforeunload', this.confirmChanges);
  },
  destroyed() {
    window.removeEventListener('beforeunload', this.confirmChanges);
  },
  methods: {
    confirmChanges(e) {
      if (this.hasUnsavedChanges) {
        e.preventDefault();
        e.returnValue = '';
      }
    },
  },
  render: () => null,
};
</script>
