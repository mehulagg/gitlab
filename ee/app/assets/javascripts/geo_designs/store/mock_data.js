import { __ } from '~/locale';

function generateRandomDateString() {
  return new Date(
    new Date(2019, 0, 1).getTime() +
      Math.random() * (new Date().getTime() - new Date(2019, 0, 1).getTime()),
  ).toString();
}

const mockData = [
  {
    id: 1,
    name: __("Zack's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: 'synced',
    last_synced_at: generateRandomDateString(),
    last_verified_at: generateRandomDateString(),
    last_checked_at: generateRandomDateString(),
  },
  {
    id: 2,
    name: __("Valery's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: 'pending',
    last_synced_at: generateRandomDateString(),
    last_verified_at: generateRandomDateString(),
    last_checked_at: generateRandomDateString(),
  },
  {
    id: 3,
    name: __("Mike's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: 'failed',
    last_synced_at: generateRandomDateString(),
    last_verified_at: generateRandomDateString(),
    last_checked_at: generateRandomDateString(),
  },
  {
    id: 4,
    name: __("Rachel's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: null,
    last_synced_at: null,
    last_verified_at: null,
    last_checked_at: null,
  },
];

export default mockData;
