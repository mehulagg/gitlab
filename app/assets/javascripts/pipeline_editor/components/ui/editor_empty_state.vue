<script>
import { GlSprintf } from '@gitlab/ui';
import { s__ } from '~/locale';

export default {
  components: {
    GlSprintf,
  },
  i18n: {
    title: s__('PipelineEditor|Optimize your workflow with CI/CD Pipelines'),
    body: s__(
      'PipelineEditor|Create a new %{codeStart}.gitlab-ci.yml%{codeEnd} file at the root of the repository to get started.',
    ),
  },
  inject: {
    emptyStateIllustrationPath: {
      default: '',
    },
  },
};
</script>
<template>
  <div class="gl-display-flex gl-flex-direction-column gl-align-items-center">
    <img :src="emptyStateIllustrationPath" data-testid="empty-state-illustration" />
    <h1 class="gl-font-size-h1">{{ $options.i18n.title }}</h1>
    <p>
      <gl-sprintf :message="$options.i18n.body" class="gl-display-inline">
        <template #code="{ content }">
          <code>{{ content }}</code>
        </template>
      </gl-sprintf>
    </p>
  </div>
</template>
