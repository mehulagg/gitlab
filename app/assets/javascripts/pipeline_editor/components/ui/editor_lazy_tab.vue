<script>
import { GlTab } from '@gitlab/ui';

/**
 * Wrapper of <gl-tab> to lazily render this tab's content
 * when shown **without dismounting it after**.
 *
 * Once the tab is selected it is permanently set as "not-lazy"
 * so it's contents are not dismounted.
 *
 * Usage:
 *
 * Same as <gl-tab>, for example:
 *
 * <gl-tabs>
 *   <editor-lazy-tab title="Tab 1"> ... </editor-lazy-tab>
 *   <editor-lazy-tab title="Tab 2"> ... </editor-lazy-tab>
 * </gl-tabs>
 */
export default {
  components: {
    GlTab,
  },
  props: {
    lazy: {
      type: Boolean,
      required: false,
      default: true,
    },
  },
  data() {
    return {
      isLazy: this.lazy,
    };
  },
  methods: {
    onClick() {
      this.isLazy = false;
    },
  },
};
</script>
<template>
  <gl-tab :lazy="isLazy" v-bind="$attrs" v-on="$listeners" @click="onClick">
    <slot v-for="slot in Object.keys($slots)" :slot="slot" :name="slot"></slot>
  </gl-tab>
</template>
