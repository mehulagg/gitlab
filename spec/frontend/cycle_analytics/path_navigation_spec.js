import { GlPath, GlDeprecatedSkeletonLoading as GlSkeletonLoading } from '@gitlab/ui';
import { mount } from '@vue/test-utils';
import Component from '~/cycle_analytics/components/path_navigation.vue';
import { transformedProjectStagePathData, selectedStage } from './mock_data';

describe('Project PathNavigation', () => {
  let wrapper = null;

  const createComponent = (props) => {
    return mount(Component, {
      propsData: {
        stages: transformedProjectStagePathData,
        selectedStage,
        loading: false,
        ...props,
      },
    });
  };

  const pathNavigationTitles = () => {
    return wrapper.findAll('.gl-path-button');
  };

  const pathNavigationItems = () => {
    return wrapper.findAll('.gl-path-nav-list-item');
  };

  const clickItemAt = (index) => {
    pathNavigationTitles().at(index).trigger('click');
  };

  beforeEach(() => {
    wrapper = createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('displays correctly', () => {
    it('has the correct props', () => {
      expect(wrapper.find(GlPath).props('items')).toMatchObject(transformedProjectStagePathData);
    });

    it('contains all the expected stages', () => {
      const html = wrapper.find(GlPath).html();

      transformedProjectStagePathData.forEach((stage) => {
        expect(html).toContain(stage.title);
      });
    });

    describe('loading', () => {
      describe('is false', () => {
        it('displays the gl-path component', () => {
          expect(wrapper.find(GlPath).exists()).toBe(true);
        });

        it('hides the gl-skeleton-loading component', () => {
          expect(wrapper.find(GlSkeletonLoading).exists()).toBe(false);
        });

        // TODO: make this test more granular
        it('matches the snapshot', () => {
          expect(wrapper.element).toMatchSnapshot();
        });

        describe('popovers', () => {
          beforeEach(() => {
            wrapper = createComponent({ stages: transformedProjectStagePathData });
          });

          it('renders popovers for all stages', () => {
            const pathItemContent = pathNavigationItems().wrappers;

            pathItemContent.forEach((stage) => {
              expect(stage.find('[data-testid="stage-item-popover"]').exists()).toBe(true);
            });
          });

          it('shows the median stage time for the first stage item', () => {
            const firstPopover = wrapper.findAll('[data-testid="stage-item-popover"]').at(0);
            expect(firstPopover.text()).toContain('Stage time (median)');
          });
        });
      });

      describe('is true', () => {
        beforeEach(() => {
          wrapper = createComponent({ loading: true });
        });

        it('hides the gl-path component', () => {
          expect(wrapper.find(GlPath).exists()).toBe(false);
        });

        it('displays the gl-skeleton-loading component', () => {
          expect(wrapper.find(GlSkeletonLoading).exists()).toBe(true);
        });
      });
    });
  });

  describe('event handling', () => {
    it('emits the selected event', () => {
      expect(wrapper.emitted('selected')).toBeUndefined();

      clickItemAt(0);
      clickItemAt(1);
      clickItemAt(2);

      expect(wrapper.emitted().selected).toEqual([
        [transformedProjectStagePathData[0]],
        [transformedProjectStagePathData[1]],
        [transformedProjectStagePathData[2]],
      ]);
    });
  });
});
