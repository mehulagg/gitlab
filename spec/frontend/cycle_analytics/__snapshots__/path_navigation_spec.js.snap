// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project PathNavigation displays correctly loading is false matches the snapshot 1`] = `
<div
  class="gl-path-nav"
  data-testid="gl-path-nav"
  style="--path-bg-color: white;"
>
  <span
    class="gl-path-fade gl-path-fade-left"
    style="display: none;"
  >
    <button
      aria-label="Scroll left"
      class="gl-clear-icon-button"
    >
      <svg
        aria-hidden="true"
        class="gl-icon s32"
        data-testid="chevron-left-icon"
        role="img"
      >
        <use
          href="#chevron-left"
        />
      </svg>
    </button>
  </span>
   
  <ul
    class="gl-path-nav-list"
  >
    <li
      class="gl-path-nav-list-item"
      id="path-6-item-0"
    >
      <button
        class="gl-path-button gl-path-active-item-indigo"
      >
        <!---->
        
        Issue
        <span
          class="gl-font-weight-normal gl-pl-2"
        >
          172800
        </span>
      </button>
       
      <div
        class="gl-popover"
        data-testid="stage-item-popover"
      >
         
        <div
          class="gl-px-4"
        >
          <div
            class="gl-display-flex gl-justify-content-space-between"
          >
            <div
              class="gl-pr-4 gl-pb-4"
            >
              
            Stage time (median)
          
            </div>
             
            <div
              class="gl-pb-4 gl-font-weight-bold"
            >
              172800
            </div>
          </div>
        </div>
         
        <div
          class="gl-px-4"
        >
          <div
            class="gl-display-flex gl-justify-content-space-between"
          >
            <div
              class="gl-pr-4 gl-pb-4"
            >
              
            Items in stage
          
            </div>
             
            <div
              class="gl-pb-4 gl-font-weight-bold"
            >
              -
            </div>
          </div>
        </div>
         
        <div
          class="gl-px-4 gl-pt-4 gl-border-t-1 gl-border-t-solid gl-border-gray-50"
        >
          <!---->
           
          <!---->
        </div>
        Issue
      </div>
    </li>
    <li
      class="gl-path-nav-list-item"
      id="path-6-item-1"
    >
      <button
        class="gl-path-button"
      >
        <!---->
        
        Plan
        <span
          class="gl-font-weight-normal gl-pl-2"
        >
          86400
        </span>
      </button>
       
      <div
        class="gl-popover"
        data-testid="stage-item-popover"
      >
         
        <div
          class="gl-px-4"
        >
          <div
            class="gl-display-flex gl-justify-content-space-between"
          >
            <div
              class="gl-pr-4 gl-pb-4"
            >
              
            Stage time (median)
          
            </div>
             
            <div
              class="gl-pb-4 gl-font-weight-bold"
            >
              86400
            </div>
          </div>
        </div>
         
        <div
          class="gl-px-4"
        >
          <div
            class="gl-display-flex gl-justify-content-space-between"
          >
            <div
              class="gl-pr-4 gl-pb-4"
            >
              
            Items in stage
          
            </div>
             
            <div
              class="gl-pb-4 gl-font-weight-bold"
            >
              -
            </div>
          </div>
        </div>
         
        <div
          class="gl-px-4 gl-pt-4 gl-border-t-1 gl-border-t-solid gl-border-gray-50"
        >
          <!---->
           
          <!---->
        </div>
        Plan
      </div>
    </li>
    <li
      class="gl-path-nav-list-item"
      id="path-6-item-2"
    >
      <button
        class="gl-path-button"
      >
        <!---->
        
        Code
        <span
          class="gl-font-weight-normal gl-pl-2"
        >
          129600
        </span>
      </button>
       
      <div
        class="gl-popover"
        data-testid="stage-item-popover"
      >
         
        <div
          class="gl-px-4"
        >
          <div
            class="gl-display-flex gl-justify-content-space-between"
          >
            <div
              class="gl-pr-4 gl-pb-4"
            >
              
            Stage time (median)
          
            </div>
             
            <div
              class="gl-pb-4 gl-font-weight-bold"
            >
              129600
            </div>
          </div>
        </div>
         
        <div
          class="gl-px-4"
        >
          <div
            class="gl-display-flex gl-justify-content-space-between"
          >
            <div
              class="gl-pr-4 gl-pb-4"
            >
              
            Items in stage
          
            </div>
             
            <div
              class="gl-pb-4 gl-font-weight-bold"
            >
              -
            </div>
          </div>
        </div>
         
        <div
          class="gl-px-4 gl-pt-4 gl-border-t-1 gl-border-t-solid gl-border-gray-50"
        >
          <!---->
           
          <!---->
        </div>
        Code
      </div>
    </li>
  </ul>
   
  <span
    class="gl-path-fade gl-path-fade-right"
    style="display: none;"
  >
    <button
      aria-label="Scroll right"
      class="gl-clear-icon-button"
    >
      <svg
        aria-hidden="true"
        class="gl-icon s32"
        data-testid="chevron-right-icon"
        role="img"
      >
        <use
          href="#chevron-right"
        />
      </svg>
    </button>
  </span>
</div>
`;
